sql/driver/conn.go

```diff
@@ -62,8 +64,39 @@ func (c *conn) Exec(stmt string, args []driver.Value) (driver.Result, error) {
 }
 
 func (c *conn) Query(stmt string, args []driver.Value) (*rows, error) {
-	// TODO(vivek): Add the args to the Call.
-	return c.send(Request{RequestHeader: RequestHeader{Session: c.session}, Sql: stmt})
+	params := make([]Datum, 0, len(args))
+	for _, arg := range args {
+		if arg == nil {
+			return nil, errors.New("Passed in a nil parameter")
```

Comment:
The spec says that Value can be nil
https://golang.org/pkg/database/sql/driver/#Value

I'm leaving it as is.